datus-clickhouse/datus_clickhouse/connector.py

```
@@ -0,0 +1,407 @@
+    def __init__(self, config: Union[ClickHouseConfig, dict]):
+        """
+        Initialize ClickHouse connector.
+
+        Args:
+            config: ClickHouseConfig object or dict with configuration
+        """
+        # Handle config object or dict
+        if isinstance(config, dict):
+            config = ClickHouseConfig(**config)
+        elif not isinstance(config, ClickHouseConfig):
+            raise TypeError(f"config must be ClickHouseConfig or dict, got {type(config)}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)